packages/esm-patient-allergies-app/src/allergies/allergies-form/allergy-form.component.tsx

```diff
@@ -66,6 +66,22 @@ const AllergyForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patientU
   const allergenTypes = [t('drug', 'Drug'), t('food', 'Food'), t('environmental', 'Environmental')];
   const severityLevels = [t('mild', 'Mild'), t('moderate', 'Moderate'), t('severe', 'Severe')];
 
+  useEffect(() => {
+    promptBeforeClosing(() => {
+      return Boolean(
+        nonCodedAllergenType || nonCodedAllergicReaction || onsetDate || selectedAllergen || severityOfWorstReaction,
+      );
+    });
+  }, [
+    promptBeforeClosing,
```

Comment:
is there an easier way of observing the form has values other than for watching on every question? This makes me concerned for the more complex forms which have dozens of questions.

Reply:
Yeah I mean ideally our forms would use some kind of abstraction that would make this easy. Like we should definitely handle this in a generic way in the actual form engine.